File: tkpdabstraction/src/main/java/com/tokopedia/abstraction/common/data/model/request/DataRequest.java
package com.tokopedia.abstraction.common.data.model.request;

import com.google.gson.annotations.SerializedName;

/**
 * Created by User on 11/8/2017.
 */

public class DataRequest<T> {
    @SerializedName("data")
    private T data;

    public DataRequest(){

    }
    public DataRequest(T data){
        this.data = data;
    }
}

File: features/recharge/common_digital/src/main/java/com/tokopedia/common_digital/common/data/request/DataRequest.kt
package com.tokopedia.common_digital.common.data.request

import com.google.gson.annotations.SerializedName

/**
 * Created by User on 11/8/2017.
 */
data class DataRequest<T> (
    @SerializedName("data")
    var data: T? = null
)